### ISSUE TITLE
Injector.provide fails with TypeError when using injection tokens

### ISSUE THREAD

**Issue by laho** [Urigo/graphql-modules#652]
Injector.provide was changed in release https://github.com/Urigo/graphql-modules/releases/tag/v0.7.10 related to issue #600.

More specifically, this line was changed to read metadata from the provider object:

https://github.com/Urigo/graphql-modules/blob/cefd7c2e5c9c1cac8928b300c6d767c9f34de252/packages/di/src/injector.ts#L128

However, when injection tokens are used this provider.provide is a string causing a `TypeError` to be thrown from the Reflect.getMetadata function. E.g.

```
providers: [{
  provide: 'MY_PROVIDER_ID',
  useClass: MyProvider
}]
```

This is the stack trace of the error thrown in server creation:
```
      at Object.getMetadata (node_modules/reflect-metadata/Reflect.js:354:23)
      at Injector.provide (node_modules/@graphql-modules/di/src/injector.ts:128:61)
      at new Injector (node_modules/@graphql-modules/di/src/injector.ts:47:14)
      at GraphQLModule.get injector [as injector] (node_modules/@graphql-modules/core/src/graphql-module.ts:472:48)
      at ModuleSessionInfo.get injector [as injector] (node_modules/@graphql-modules/core/src/module-session-info.ts:22:24)
      at new ModuleSessionInfo (node_modules/@graphql-modules/core/src/module-session-info.ts:6:10)
      at _sessionContext$Map.set.Promise (node_modules/@graphql-modules/core/src/graphql-module.ts:1287:43)
      at GraphQLModule._cache.contextBuilder (node_modules/@graphql-modules/core/src/graphql-module.ts:1276:13)
      at Promise.all.selfImports.map.module (node_modules/@graphql-modules/core/src/graphql-module.ts:1283:54)
          at Array.map (<anonymous>)
      at _sessionContext$Map.set.Promise (node_modules/@graphql-modules/core/src/graphql-module.ts:1283:33)
      at _cache.contextBuilder (node_modules/@graphql-modules/core/src/graphql-module.ts:1276:13)
```

**Comment by laho** [comment]
Seems to fix the issue. 👍

**Comment by ardatan** [comment]
Fixed in 0.7.11!
